src/main/java/org/folio/services/kafka/topic/KafkaAdminClientService.java

```diff
@@ -78,24 +68,19 @@ public KafkaAdminClientService(Supplier<KafkaAdminClient> clientFactory) {
   private Future<Void> createKafkaTopics(String tenantId, String environmentName,
                                          KafkaAdminClient kafkaAdminClient) {
 
-    final List<NewTopic> expectedTopics = readTopics()
+    final List<NewTopic> topics = readTopics()
       .map(topic -> qualifyName(topic, environmentName, tenantId))
+      .map(topic -> topic.setReplicationFactor(getReplicationFactor()))
       .collect(Collectors.toList());
 
-    return kafkaAdminClient.listTopics().compose(existingTopics -> {
-      final List<NewTopic> topicsToCreate = expectedTopics.stream()
-        .filter(topic -> !existingTopics.contains(topic.getName()))
-        .map(topic -> topic.setReplicationFactor(getReplicationFactor()))
-        .collect(Collectors.toList());
-
-      if (topicsToCreate.isEmpty()) {
-        log.info("All topics already exists, skipping creation");
-        return succeededFuture();
-      }
-
-      log.info("Creating topics {}", topicsToCreate);
-      return kafkaAdminClient.createTopics(topicsToCreate);
-    });
+    return kafkaAdminClient.createTopics(topics)
+      .recover(x -> {
+        if (x instanceof org.apache.kafka.common.errors.TopicExistsException) {
+          return Future.succeededFuture();
```

Comment:
Could we log that this error has occurred?

As I understand it, this change introduced the chance that the creation of one topic fails, leaving other topics not created and yet the Tenant API request will succeed stating that the module should work for the tenant.

Reply:
Kafka waits until all topics have been processed before throwing a TopicExistsException. Only if no other topic caused some other exception a TopicExistsException is thrown.
https://github.com/apache/kafka/blob/3.0.0/trogdor/src/main/java/org/apache/kafka/trogdor/common/WorkerUtils.java#L215

Reply:
@julianladisch Thank you for the explanation and link

Reply:
Thx @julianladisch . I made a test for this, although not a bullet proof one. I'll add a log msg message. level? log? warn?

Reply:
info.